build-system/pr-check.js

```diff
@@ -126,7 +126,9 @@ function isBuildSystemFile(filePath) {
       !isFlagConfig(filePath) &&
       // Exclude visual diff files from build-system since we want it to trigger
       // visual diff tests.
-      !isVisualDiffFile(filePath);
+      !isVisualDiffFile(filePath))
+      // OWNERS.yaml files should trigger build system to run tests
+      || path.basename(filePath) === 'OWNERS.yaml';
```

Comment:
Since `gulp ava` will indeed run your new tests, I think all you need is an additional check to make sure `CODEOWNERS` is updated too.